packages/amplify-graphql-relational-transformer/src/__tests__/amplify-graphql-has-many-transformer.test.ts

```diff
@@ -850,3 +851,78 @@ describe('Pre Processing Has Many Tests', () => {
     expect(hasGeneratedField(updatedSchemaDoc, 'Post', 'blogPostsFieldValue', 'String')).toBeTruthy();
   });
 });
+
+describe('@hasMany connection field nullability tests', () => {
+  const featureFlags: FeatureFlagProvider = {
+    getBoolean: (value: string, defaultValue: boolean): boolean => {
+      if (value === 'respectPrimaryKeyAttributesOnConnectionField') {
+        return true;
+      }
+      return defaultValue;
+    },
+    getNumber: jest.fn(),
+    getObject: jest.fn(),
+  };
+  test('Should not affect the nullability of connection fields of the other side when the @hasMany field is non-nullable', () => {
+    const inputSchema = `
+      type Todo @model {
+        todoid: ID! @primaryKey(sortKeyFields:["name"])
+        name: String!
+        title: String!
+        priority: Int
+        tasks: [Task]! @hasMany
+      }
+      type Task @model {
+        taskid: ID! @primaryKey(sortKeyFields:["name"])
+        name: String!
+        description: String
+      }
+    `;
+    const transformer = new GraphQLTransform({
+      featureFlags,
+      transformers: [new ModelTransformer(), new PrimaryKeyTransformer(), new HasManyTransformer()],
+    });
+  
+    const out = transformer.transform(inputSchema);
+    expect(out).toBeDefined();
+    const schema = parse(out.schema);
+    validateModelSchema(schema);
+    const connectionFieldName1 = 'todoTasksTodoid';
+    const connectionFieldName2 = 'todoTasksName';
+    //Type definition
+    const objType = schema.definitions.find((def: any) => def.name && def.name.value === 'Task') as any;
+    expect(objType).toBeDefined();
+    const relatedField1 = objType.fields.find((f: any) => f.name.value === connectionFieldName1);
+    expect(relatedField1).toBeDefined();
+    expect(relatedField1.type.kind).toBe(Kind.NAMED_TYPE);
+    expect(relatedField1.type.name.value).toBe('ID');
+    const relatedField2 = objType.fields.find((f: any) => f.name.value === connectionFieldName2);
+    expect(relatedField2).toBeDefined();
+    expect(relatedField2.type.kind).toBe(Kind.NAMED_TYPE);
+    expect(relatedField2.type.name.value).toBe('String');
+    //Create Input
+    const createInput = schema.definitions.find((def: any) => def.name && def.name.value === 'CreateTaskInput') as any;
+    expect(createInput).toBeDefined();
+    expect(createInput.fields.length).toEqual(5);
+    const createInputConnectedField1 = createInput.fields.find((f: any) => f.name.value === connectionFieldName1);
+    expect(createInputConnectedField1).toBeDefined();
+    expect(createInputConnectedField1.type.kind).toBe(Kind.NAMED_TYPE);
```

Comment:
Double check this test if we should return optional types for connected fields in case of a unidirectional `@hasMany` attached to a required field.